guides/plugins/plugins/storefront/add-scss-variables.md

```diff
@@ -0,0 +1,212 @@
+    {
+        /** @var string $configExampleField */
+        $configPluginHeaderBgColor = $this->systemConfig->get('SwagBasicExample.config.sassPluginHeaderBgColor', $event->getSalesChannelId());
+
+        // pass the value from `configPluginHeaderBgColor` to `addVariable`
+        $event->addVariable('sass-plugin-header-bg-color', $configPluginHeaderBgColor);
+    }
+}
+```
+{% endtab %}
+{% tab title="<plugin root>/src/Resources/config/services.xml" %}
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    
+    <services>
+        <service id="Swag\BasicExample\Subscriber\ThemeVariablesSubscriber">
+            <!-- add argument `SystemConfigService` -->
+            <argument type="service" id="Shopware\Core\System\SystemConfig\SystemConfigService"/>
+            <tag name="kernel.event_subscriber"/>
+        </service>
+    </services>
+</container>
+```
+{% endtab %}
+{% endtabs %}
+
+* The `SystemConfigService` provides a `get()` method where you can access the configuration structure in the first parameter with a dot notation syntax like `SwagBasicExample.config.fieldName`. The second parameter is the sales-channel `id`. With this `id` the config fields can be accessed for each sales-channel.
+* You can get the sales-channel id through the getter `getSalesChannelId()` of the `ThemeCompilerEnrichScssVariablesEvent`.
+* Now your sass variables can have different values in each sales-channel.
+
+### All config fields as SCSS variables
+
+Adding config fields via `$event->addVariable()` individually for every field may be a bit cumbersome in some cases. You could also loop over all config fields and call `addVariable()` for each config field. But this depends on your use case.
+
+{% code title="<plugin root>/src/Subscriber/ThemeVariablesSubscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+// ...
+use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter;
+
+class ThemeVariablesSubscriber implements EventSubscriberInterface
+{
+    // ...
+
+    public function onAddVariables(ThemeCompilerEnrichScssVariablesEvent $event)
+    {
+        $configFields = $this->systemConfig->get('SwagBasicExample.config', $event->getSalesChannelId());
+
+        foreach($configFields as $key => $value) {
+            // Convert `customVariableName` to `custom-variable-name`
+            $variableName = str_replace('_', '-', (new CamelCaseToSnakeCaseNameConverter())->normalize($key));
+
+            $event->addVariable($variableName, $value);
+        }
+    }
+}
+```
+{% endcode %}
+
+To avoid camelCase variable names when reading from the `config.xml` we recommend using the `CamelCaseToSnakeCaseNameConverter` to format the variable before adding it.
```

Comment:
I suggest to refer to kebab case so the format that is the result of combining the camelcase to snake and str_replace gets a name.

Reply:
@JoshuaBehrens - tried it, unfortunately I couldn't get it to work.

Reply:
@raknison what do you mean you could not get it to work? What did you try? :D I thought about naming at least the variable $kebabCased or similar

Reply:
@JoshuaBehrens - now i have understood and changed it. should be clearer now.

Reply:
🥙